script.js

```diff
@@ -1,12 +1,105 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+let arrayOfEpisodes = getAllEpisodes();
+let mainEl = document.querySelector("main");
+let searchBar = document.querySelector("#searchInput");
+let displayNumberOfEpisodes = document.querySelector("#displayEpisodesText");
+let selectEl = document.querySelector("#selectMenu");
+
+//search bar
+searchBar.addEventListener("keyup", (e) => {
+  const searchString = e.target.value.toLowerCase();
+  const filteredEpisodes = arrayOfEpisodes.filter((episode) => {
+    return (
+      episode.summary.toLowerCase().includes(searchString) ||
+      episodeCode(episode).toLowerCase().includes(searchString) ||
+      episode.name.toLowerCase().includes(searchString)
+    );
+  });
+  loadEpisodes(filteredEpisodes);
+  displayNumberOfEpisodes.innerText = `Displaying ${filteredEpisodes.length}/${arrayOfEpisodes.length} episodes`;
+  selectMenu(filteredEpisodes);
+});
+
+//add functionality to select menu
+selectEl.addEventListener("change", (e) => {
+  window.location = `#${e.target.value}`;
+  let allEpisodes = document.getElementsByClassName("episodeBox");
+  let highlightedEl = document.getElementById(e.target.value);
+  highlightedEl.style.border = "solid";
+  for (let i = 0; i < allEpisodes.length; i++) {
+    if (allEpisodes[i] !== highlightedEl) {
+      allEpisodes[i].style.border = null;
+    }
+  }
+});
+
+//Formatting name of an episode
+function episodeCode(obj) {
+  let seasonNumber = obj.season < 10 ? `0${obj.season}` : `${obj.season}`;
+  let episodeNumber = obj.number < 10 ? `0${obj.number}` : `${obj.number}`;
+  return `S${seasonNumber}E${episodeNumber}`;
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+//add options in select menu
+const selectMenu = (Episodes) => {
+  const select = Episodes.map((episode) => {
+    return `
+    <option value="${episode.id}">   
+      ${episodeCode(episode)} - ${episode.name}
+    </option>;
+`;
+  }).join("");
+  selectEl.innerHTML = select;
+};
+
+//add all episodes to the page
+function loadEpisodes(Episodes) {
+  const htmlString = Episodes.map((episode) => {
+    return `
+    <div id=${episode.id} class="episodeBox">
```

Comment:
You may want to consider using `document.createElement()` instead of writing out the html manually